lib/Migration/Version5000Date20211025124248.php

```diff
@@ -192,17 +201,18 @@ protected function normalizeKey(string $prefixedKey): string {
 		throw new \RuntimeException('Invalid regex pattern');
 	}
 
+	/** @return list<int> */
 	protected function fetchPrefixes(): array {
 		$q = $this->dbc->getQueryBuilder();
 		$q->select('configvalue')
 			->from('appconfig')
 			->where($q->expr()->eq('appid', $q->createNamedParameter('user_saml')))
 			->andWhere($q->expr()->eq('configkey', $q->createNamedParameter('providerIds')));
 
-		$r = $q->executeQuery();
+		$r = $q->execute();
 		$prefixes = $r->fetchOne();
 		if ($prefixes === false) {
-			return [];
+			return [1]; // 1 is the default value for providerIds
```

Comment:
@CarlSchwan is it this part that broke for you when there was only one entry left, but  no  providerIds field present in the old appconfig? For when there is really no configuration it does not need to run at all. I would then add a check in `postSchemaChange()` to continue if `$configKeys` is empty and save readConfiguration (if only to ensure that in the DB query `cfgKeys` is not assigend an empty list and the engine may decide to error out – did not check it does, but not trying is not provoking).

Reply:
In https://github.com/nextcloud/user_saml/blob/d1f5a8314a618565237df98d1456f0ad0303165a/lib/SAMLSettings.php#L66 and https://github.com/nextcloud/user_saml/blob/d1f5a8314a618565237df98d1456f0ad0303165a/lib/Settings/Admin.php#L58 we are assuming that providerIds is by default `1`.

When I configured a SAML provider with master, my provider was properly configured in the UI and database but the providerIds key wasn't and because the default is 1, this worked (ok I was hit by https://github.com/nextcloud/user_saml/issues/428 but that's hopefully another issue :sweat_smile: ).